### PR TITLE
fix: Bubble up errors from Rust dependency

### PR DESCRIPTION
**Pull request by tmathern**

### CODE REVIEW

File: js-src/lib/error.ts [hunk 1]
```diff
@@ -27,6 +27,7 @@ export class SigningError extends Error {
   constructor(options?: ErrorOptions) {
     super(`Signing error`, options);
     this.name = this.constructor.name;
+    this.message = options?.cause?.toString() || "";
```
Comment (tmathern) [inline js-src/lib/error.ts:30]: Error text in js/ts goes in `message`

File: src/sign/remote.rs [hunk 2]
```diff
@@ -87,15 +89,30 @@ impl RemoteSigner {
...
-                        ))),
-                    })?;
+                      Ok(value) => {
+                          Ok(Ok(value
+                              .downcast_or_throw::<JsNumber, _>(&mut cx)?
+                              .value(&mut cx)))
+                      }
+                      // call to JS provided `reserveSize` fn threw an error.
+                      // we catch the neon `Throw` error and map it to a cp2a_node::Error.
+                      Err(err) => {
```
Comment (tmathern) [inline src/sign/remote.rs:99]: To log to the js console:

```
//*
// logs the error to the JS console (can be useful for debugging)
let err = err.downcast_or_throw::<JsObject, _>(&mut cx)?;
let console: Handle<JsObject> = cx.global("console")?;
let log: Handle<JsFunction> = console.get(&mut cx, "log")?;
log.call_with(&cx).arg(err).exec(&mut cx)?;
// */
```

File: tests/sign.test.ts [hunk 3]
```diff
@@ -657,6 +657,6 @@ describe('sign()', () => {
         asset,
         manifest,
       });
-    }).rejects.toThrow(/Signing error/);
+    }).rejects.toThrow("C2pa(OtherError(RemoteSign)): An error was thrown by RemoteSigner#sign");
```
Comment (tmathern) [inline tests/sign.test.ts:660]: Because we have more details now - is it OK to change?

File: tests/sign.test.ts [hunk 4]
```diff
@@ -627,7 +627,7 @@ describe('sign()', () => {
           asset,
           manifest,
         });
-      }).rejects.toThrow(/Signing error/);
+      }).rejects.toThrow("RemoteReserveSize(\"error in reserveSize fn: Error: reserveSize\"): Fetching reserve size failed: error in reserveSize fn: Error: reserveSize");
```
Comment (tmathern) [inline tests/sign.test.ts:630]: Because we have more details now - is it OK to change?

**Review by dyro — APPROVED** [review]
Looks clean!

**Pull request merged**